server/helpers/youtube-dl/youtube-dl-base-builder.ts

```diff
@@ -0,0 +1,168 @@
+      omission: ' […]'
+    })
+  }
+
+  public buildOriginallyPublishedAt (obj: any) {
+    let originallyPublishedAt: Date = null
+
+    const uploadDateMatcher = /^(\d{4})(\d{2})(\d{2})$/.exec(obj.upload_date)
+    if (uploadDateMatcher) {
+      originallyPublishedAt = new Date()
+      originallyPublishedAt.setHours(0, 0, 0, 0)
+
+      const year = parseInt(uploadDateMatcher[1], 10)
+      // Month starts from 0
+      const month = parseInt(uploadDateMatcher[2], 10) - 1
+      const day = parseInt(uploadDateMatcher[3], 10)
+
+      originallyPublishedAt.setFullYear(year, month, day)
+    }
+
+    return originallyPublishedAt
+  }
+
+  public buildAvailableUrl (obj: any) {
+    const urls: string[] = []
+
+    if (obj.url) urls.push(obj.url)
+    if (obj.urls) {
+      if (Array.isArray(obj.urls)) urls.push(...obj.urls)
+      else urls.push(obj.urls)
+    }
+
+    const formats = Array.isArray(obj.formats)
+      ? obj.formats
+      : []
+
+    for (const format of formats) {
+      if (!format.url) continue
+
+      urls.push(format.url)
+    }
+
+    const thumbnails = Array.isArray(obj.thumbnails)
+      ? obj.thumbnails
+      : []
+
+    for (const thumbnail of thumbnails) {
+      if (!thumbnail.url) continue
+
+      urls.push(thumbnail.url)
+    }
+
+    if (obj.thumbnail) urls.push(obj.thumbnail)
+
+    for (const subtitleKey of Object.keys(obj.subtitles || {})) {
+      const subtitles = obj.subtitles[subtitleKey]
+      if (!Array.isArray(subtitles)) continue
+
+      for (const subtitle of subtitles) {
+        if (!subtitle.url) continue
+
+        urls.push(subtitle.url)
+      }
+    }
+
+    return urls.filter(u => u && isUrlValid(u))
+  }
+
+  public getTags (tags: string[]) {
+    if (Array.isArray(tags) === false) return []
+
+    return tags
+      .filter(t => t.length < CONSTRAINTS_FIELDS.VIDEOS.TAG.max && t.length > CONSTRAINTS_FIELDS.VIDEOS.TAG.min)
+      .map(t => t.normalize())
+      .slice(0, 5)
+  }
+
+  public getLicence (licence: string) {
+    if (!licence) return undefined
+
+    if (licence.includes('Creative Commons Attribution')) return 1
+
+    for (const key of Object.keys(VIDEO_LICENCES)) {
+      const peertubeLicence = VIDEO_LICENCES[key]
+      if (peertubeLicence.toLowerCase() === licence.toLowerCase()) return parseInt(key, 10)
+    }
+
+    return undefined
+  }
+
+  public getCategory (categories: string[]) {
+    if (!categories) return undefined
+
+    const categoryString = categories[0]
+    if (!categoryString || typeof categoryString !== 'string') return undefined
+
+    if (categoryString === 'News & Politics') return 11
+
+    for (const key of Object.keys(VIDEO_CATEGORIES)) {
+      const category = VIDEO_CATEGORIES[key]
+      if (categoryString.toLowerCase() === category.toLowerCase()) return parseInt(key, 10)
+    }
+
+    return undefined
+  }
+
+  public getLiveStatus (status: string) {
+    return VIDEO_IMPORT_LIVE_STATUSES[status] ? status : VIDEO_IMPORT_LIVE_STATUSES.not_live
```

Comment:
I'm not sure to understand the purpose of translating the internal youtube status to plain english?

Reply:
Yeah me neither 😅

Was still getting to know the codebase, it's the first piece of code modified like a week ago.

Will modify to simply use direct enum values.

Reply:
Simplified the enum.